cql3/expr/to_restriction.cc

```diff
@@ -20,15 +20,246 @@
  */
 
 #include "expression.hh"
+#include "cql3/statements/request_validations.hh"
+#include "cql3/restrictions/single_column_restriction.hh"
+#include "cql3/prepare_context.hh"
+#include "types/list.hh"
+#include <seastar/util/defer.hh>
+#include "exceptions/unrecognized_entity_exception.hh"
 
 namespace cql3 {
 namespace expr {
-::shared_ptr<restrictions::restriction> to_restriction(binary_operator& binop,
+namespace {
+logging::logger expr_to_restriction_logger("cql3_expr_to_restriction");
+
+bool is_legal_relation_for_non_frozen_collection(oper_t oper, bool is_lhs_col_indexed) {
+    return oper == oper_t::CONTAINS_KEY || oper == oper_t::CONTAINS || (oper == oper_t::EQ && is_lhs_col_indexed);
+}
+
+bool oper_is_slice(oper_t oper) {
+    switch(oper) {
+        case oper_t::LT:
+        case oper_t::LTE:
+        case oper_t::GT:
+        case oper_t::GTE:
+            return true;
+        default:
```

Comment:
It's better to list everything and avoid `default:` to avoid surprises when something new is added.